WireDomain/Sources/WireDomain/Repositories/Message/MessageLocalStore.swift

```diff
@@ -365,24 +440,41 @@ public final class MessageLocalStore: MessageLocalStoreProtocol {
         }
     }
 
-    // swiftlint:disable:next todo_requires_jira_link
-    // TODO: Use UserLocalStore when related PRs are merged.
+    private func fetchUsers(
+        userIDs: [(id: UUID, domain: String?)]
+    ) async -> Set<ZMUser> {
+        await withTaskGroup(of: ZMUser.self) { taskGroup in
+            for userID in userIDs {
+                taskGroup.addTask { [self] in
```

Comment:
Not a request for change but just a thought:

I think there is probably unnecessary overhead with awaiting, locking in core data etc here. To avoid this we I think we should have a `UserLocalStore.fetchOrCreateUsers(userIDs: [(id: UUID, domain: String?)]) -> [ZMUser]` and have everything done within a single perform.

When wrapping core data it is probably more performant that our methods accept arrays of values I guess.

Reply:
Good point, done in https://github.com/wireapp/wire-ios/pull/2175/commits/ceef40d35b39388b3a83ca4ef2aa175ba3b750db thx!